### PR TITLE
drivers/soft_spi: move to drivers_periph group

### PR DESCRIPTION
**Pull request by aabadie**
### Contribution description

This PR moves the soft_spi driver doxygen group to the drivers_periph group otherwise it appears at the top level drivers group in the documentation, e.g at the same level as Network, Actuators, Sensors, etc.

Even if the drivers_periph is more related to MCU peripherals and soft_spi is just built on top of GPIOs, I think it makes more sense to have it here, close to the hardware SPI driver. And I didn't find a better place for it.

The added empty line fixes an issue with the brief display in the documentation: without it no newline is added to the generated html and the beginning of the detailed description is displayed with the brief description.

### Issues/PRs references

None

### CODE REVIEW

**Comment by aabadie** [comment]
This one can be merged at any time @miri64

**Comment by PeterKietzmann** [comment]
How about `drivers_periph_spi`?

**Comment by aabadie** [comment]
> How about drivers_periph_spi?

I didn't think at this option. Looking at the doxygen documentation, this would move SPI deeper in the modules tree. Is this better ? I don't think so.

We could also create a new `drivers_soft_periph`, with `drivers_soft_spi` and one day `drivers_soft_uart` or `drivers_soft_i2c`. `drivers_soft_periph` will be at the same level as others top level `drivers` groups (beside `drivers_storage`, etc). What do you think ?

**Comment by miri64** [comment]
> We could also create a new drivers_soft_periph, with drivers_soft_spi inside and maybe one day drivers_soft_uart or drivers_soft_i2c. drivers_soft_periph will be at the same level as others top level drivers groups (beside drivers_periph, drivers_storage, etc). What do you think ?

I like that.

**Comment by miri64** [comment]
> We could also create a new drivers_soft_periph, with drivers_soft_spi inside and maybe one day drivers_soft_uart or drivers_soft_i2c. drivers_soft_periph will be at the same level as others top level drivers groups (beside drivers_periph, drivers_storage, etc). What do you think ?
We could also create a new drivers_soft_periph, with drivers_soft_spi inside and maybe one day drivers_soft_uart or drivers_soft_i2c. drivers_soft_periph will be at the same level as others top level drivers groups (beside drivers_periph, drivers_storage, etc). What do you think ?


I like that.

**Comment by aabadie** [comment]
> I like that.

I'll change that.

**Comment by aabadie** [comment]
@miri64, done !

File: drivers/doc.txt [hunk 1]
```diff
@@ -43,3 +43,9 @@
  * @ingroup     drivers
  * @brief       Drivers for storage devices
  */
+
+ /**
+ * @defgroup    drivers_soft_periph Soft Peripheral Driver Interface
+ * @ingroup     drivers
+ * @brief       Software emulated peripheral driver interface for UART, SPI, etc
```
Comment (miri64) [inline drivers/doc.txt]: Maybe add a link to the actual peripheral driver interfaces?

```
Software emulated @ref drivers_periph for UART, SPI, etc
```

**Comment by aabadie** [comment]
> Minor enhancement proposal

Applied and directly amended.

**Pull request merged**